synapse/telepath.py

```diff
@@ -103,15 +106,30 @@ def zipurl(info):
 
     return url
 
+def modurl(url, **info):
+    if isinstance(url, str):
+        urlinfo = chopurl(url)
+        for k, v in info.items():
+            urlinfo[k] = v
```

Comment:
Should this be using setdefault() to avoid smashing over existing usernames?

Reply:
Nope, the whole point of this API is to smash over the existing keys.